test/test_mixed_null_assertions.cpp

```diff
@@ -78,3 +78,207 @@ TEST(List_Mixed_do_insert)
     list.insert_null(0);
     list.insert(0, Mixed("hello world"));
 }
+
+TEST(Mixed_List_unresolved_as_null)
+{
+    Group g;
+    auto t = g.add_table("foo");
+    t->add_column_list(type_Mixed, "mixeds");
+    auto obj = t->create_object();
+    auto obj1 = t->create_object();
+
+    auto list = obj.get_list<Mixed>("mixeds");
+
+    list.insert_null(0);
+    list.insert(1, Mixed{"test"});
+    list.insert(2, obj1);
+    obj1.invalidate();
+
+    CHECK(list.size() == 3);
+
+    {
+        // find all mixed nulls or unresolved link should work the same way
+        std::vector<size_t> found;
```

Comment:
I don't this this is what we agreed. There should only be one version of find. And it should just search using the raw values. So I don't think there should be any change to search.

Reply:
> I don't this this is what we agreed. There should only be one version of find. And it should just search using the raw values. So I don't think there should be any change to search.

I think we need to decide if:

`find_all()` for `Lst<Mixed>` and `Set<Mixed>` must find all the nulls or not. 
It seems we have agreed to:
1. `List<Mixed>::find_all(null)` finds all the nulls (nulls and unresolved)
2. `Set<Mixed>::find_all(null)` finds just one null

`Set<Mixed>::erase_null() ` should erase only the actual null values (not the unresolved links).

do `Lst<Mixed>::sort `and `Lst<Mixed>::distinct ` need a special handling, since we need to call a more expensive method get() rather than m_tree->get() for comparing things?

Reply:
Brought the code to what I described above. Have another look

Reply:
Ok, I agree that this matches what we discussed for list.

Reply:
OK, so what's the problem with `Set<Mixed>`?